lib/src/command/token_add.dart

```diff
@@ -61,4 +63,33 @@ class TokenAddCommand extends PubCommand {
       throw ApplicationException('Token is not provided within 15 minutes.');
     }
   }
+
+  Future<void> _addTokenFromStdin(Uri hostedUrl) async {
+    final token = await stdinPrompt('Enter secret token:')
+        .timeout(const Duration(minutes: 15));
+    if (token.isEmpty) {
+      usageException('Token is not provided.');
+    }
+
+    tokenStore.addCredential(Credential.token(hostedUrl, token));
+    log.message(
+      'Requests to $hostedUrl will now be authenticated using the secret '
+      'token.',
+    );
+  }
+
+  Future<void> _addTokenFromEnv(Uri hostedUrl) async {
+    tokenStore.addCredential(Credential.env(hostedUrl, envVar));
```

Comment:
Here we can check the format eg:
```suggestion
    if (envVar.isEmpty) {
      throw DataException('Cannot use the empty string as --env-var');
    }
    tokenStore.addCredential(Credential.env(hostedUrl, envVar));
```